src/types/lib/index.ts

```diff
@@ -99,6 +100,11 @@ export type DeclareContractPayload = {
   compiledClassHash?: string;
 };
 
+/**
+ * DeclareContractPayload with classHash or contract defined
+ */
+export type ContractClassIdentifier = DeclareContractPayload | { classHash: string };
```

Comment:
Why an object containing the classHash? Seems complicated.
I think that the easiest way for the user is something like this :
```typescript
export type ContractClassIdentifier = DeclareContractPayload | BigNumberish;
```

Reply:
Because it would be ambiguous.
Class can be identified by classHash or by contract, in case of BigNumberish it can be both and non-specific.

Reply:
For me, a class is only identified by its hash, and nothing else.

Reply:
And if you have the address of an instance of a class, you have already the answer of question `is classDeclared`? 
It's more than weird to ask if a parent is existing if you have already one of its children....

Reply:
> And if you have the address of an instance of a class, you have already the answer of the question `is classDeclared`. It's more than weird to ask if a parent is existing if you have already one of its children....

Agree, and this does not contain an address. What address are you referring to?

Reply:
It does not deal with **Contract instance** nor addresses, just with **Contract Class** aka. definition classHash

**ContractClassIdentifier is** 

```ts
{
  contract: CompiledContract | string;
  classHash?: string;
  casm?: CompiledSierraCasm;
  compiledClassHash?: string;
};
```
In this case it covers cases for:
1. identifying class from its source code by computing classHash from contract
2. optional zero computing and classHash is already provided
3. less computing by providing optional params for classHash

```ts
or 
{ classHash: string }
```
This case cover providing non-optional classHash

So that in the end, you can get classHash which is in essence ContractClassIdentifier. 

Changing it to 
```ts
export type ContractClassIdentifier = DeclareContractPayload | BigNumberish;
```
In this case, BigNumberish represents classHash but in the source, we used mixed cases sometimes string sometimes BigNumberish.

Solutions:
1.
Another case is that we set ` | { classHash: BigNumberish }`.
This case is less ambiguous and I think we can do it.

2. create `type ClassHashIdentifier: BigNumberish`
and than do:
```ts
export type ContractClassIdentifier = DeclareContractPayload | ClassHashIdentifier;
```

Reply:
> For me, a class is only identified by its hash, and nothing else.

Yes but you can provide that classHash or you can provide source code and we can compute classHash from it.
So both source code and classHash are identifiers